src/core_init_atmosphere/mpas_init_atm_static.F

```diff
@@ -306,66 +323,127 @@ subroutine init_atm_static(mesh, dims, configs)
        call mpas_log_write('Please correct the namelist.', messageType=MPAS_LOG_CRIT)
  end select
 
- do jTileStart = 1,20401,ny-6
-    jTileEnd = jTileStart + ny - 1 - 6
-
-    do iTileStart=1,42001,nx-6
-       iTileEnd = iTileStart + nx - 1 - 6
-       write(fname,'(a,i5.5,a1,i5.5,a1,i5.5,a1,i5.5)') trim(geog_data_path)//trim(geog_sub_path), &
-                    iTileStart,'-',iTileEnd,'.',jTileStart,'-',jTileEnd
-       call mpas_log_write(trim(fname))
-       call mpas_f_to_c_string(fname, c_fname)
-
-       call read_geogrid(c_fname,rarray_ptr,nx,ny,nz,isigned,endian, &
-                         scalefactor,wordsize,istatus)
-       call init_atm_check_read_error(istatus, fname)
-
-       iPoint = 1
-       do j=4,ny-3
-       do i=4,nx-3
-          lat_pt = start_lat + (jTileStart + j - 5) * 0.0083333333
-          lon_pt = start_lon + (iTileStart + i - 5) * 0.0083333333
-          lat_pt = lat_pt * PI / 180.0
-          lon_pt = lon_pt * PI / 180.0
-
-          iPoint = nearest_cell(lat_pt,lon_pt,iPoint,nCells,maxEdges, &
-                                nEdgesOnCell,cellsOnCell, &
-                                latCell,lonCell)
-
-          !
-          ! For all but the outermost boundary cells, we can safely assume that the nearest
-          ! model grid cell contains the pixel (else, a different cell would be nearest)
-          !
-          if (bdyMaskCell(iPoint) < nBdyLayers) then
-             ter(iPoint) = ter(iPoint) + rarray(i,j,1)
-             nhs(iPoint) = nhs(iPoint) + 1
-
-          ! For outermost boundary cells, additional work is needed to verify that the pixel
-          ! actually lies within the nearest cell
-          else
-             zPixel = sphere_radius * sin(lat_pt)                  ! Model cell coordinates assume a "full" sphere radius
-             xPixel = sphere_radius * cos(lon_pt) * cos(lat_pt)    ! at this point, so we need to ues the same radius
-             yPixel = sphere_radius * sin(lon_pt) * cos(lat_pt)    ! for source pixel coordinates
+ ierr = mgr % init(trim(config_geog_data_path)//trim(geog_sub_path))
+ if (ierr /= 0) then
+    call mpas_log_write('Error initializing ter Geotile Manager', messageType=MPAS_LOG_CRIT)
```

Comment:
I don't have any suggestions at the moment, but are there ways that this error message could be made to be more helpful or understandable by a user? I suspect most users would have no idea what it means to not be able to initialize the "ter Geotile Manager", nor why that might have occurred.

Reply:
The `init` function of the `geotile_manager` can only produce a non-zero return value in two circumstances: [inability to find an `index` file in the given directory](https://github.com/MPAS-Dev/MPAS-Model/blob/a667cd3108cae728a977daa8bb7eef7811d31aa5/src/core_init_atmosphere/mpas_geotile_manager.F#L129-L133) and [if `parse_geoindex` returns an error](https://github.com/MPAS-Dev/MPAS-Model/blob/a667cd3108cae728a977daa8bb7eef7811d31aa5/src/core_init_atmosphere/mpas_geotile_manager.F#L139-L143). Both of which produce their own log messages. As well, the `parse_geoindex` module also produces its own log message. So this error is a catch-all for all of those possible errors and really only serves as a call kill the model (via `MPAS_LOG_CRIT`). 

So it would hard to get more specific with the error, without repeated whats already been logged, but I agree that the error message might be a bit confusing and could be better. How does the following sound?

`'Error occurred when initializing the interpolation of terrain height (ter)'`

Or,

`'Error occurred when initializing the interpolation of terrain height (ter) from '//trim(directory)`

Where `directory` would be the complete path to the data-set.

Let me know what your thoughts are or if you have any other ideas!